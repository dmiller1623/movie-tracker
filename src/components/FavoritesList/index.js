import React from 'react';
import MovieCard from '../../containers/MovieCard';
import NavBar from '../../containers/NavBar';

const FavoritesList = ({ movies, favorites }) => {
  const displayedMovies = favorites.map( (favorite, index) => {
    const movie = movies.find( movie => movie.movie_id === favorite);
    return <MovieCard {...movie} key={index} starColor={'#ffd24d'}/>
  })
  return (
    <div>
    <NavBar />
    <section>
      {displayedMovies}
    </section>
    </div>
  )
}

export default FavoritesList;