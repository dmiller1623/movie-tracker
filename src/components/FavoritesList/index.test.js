import React from 'react';
import FavoritesList from '../FavoritesList';
import { shallow } from 'enzyme';

describe('FavoritesList', ()=> {
  const mockMovies = [
    {movie_id: 2, title: "Hotori - The Simple Wish for Joy", poster_path: "/qIODP7buZlHBbbs6PNlQxb4orHo.jpg", release_date: "2005-08-28", vote_average: 10},
    {movie_id: 3, title: "Big Meat Eater", poster_path: "/gimNITXaKUevTsbXNUFEI6bCqro.jpg", release_date: "1982-04-30", vote_average: 10}
  ];
  it('should match the snapshot if there are favorites to display', () => {
    
    let wrapper = shallow(<FavoritesList movies={mockMovies} favorites={[2]}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if there are no favorites to display', () => {
    
    let wrapper = shallow(<FavoritesList movies={mockMovies} favorites={[]}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
