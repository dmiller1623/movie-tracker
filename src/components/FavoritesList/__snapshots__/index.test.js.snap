// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList should match the snapshot if there are favorites to display 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesList
    favorites={
        Array [
            2,
          ]
    }
    movies={
        Array [
            Object {
              "movie_id": 2,
              "poster_path": "/qIODP7buZlHBbbs6PNlQxb4orHo.jpg",
              "release_date": "2005-08-28",
              "title": "Hotori - The Simple Wish for Joy",
              "vote_average": 10,
            },
            Object {
              "movie_id": 3,
              "poster_path": "/gimNITXaKUevTsbXNUFEI6bCqro.jpg",
              "release_date": "1982-04-30",
              "title": "Big Meat Eater",
              "vote_average": 10,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(MovieCard)
          movie_id={2}
          poster_path="/qIODP7buZlHBbbs6PNlQxb4orHo.jpg"
          release_date="2005-08-28"
          starColor="#ffd24d"
          title="Hotori - The Simple Wish for Joy"
          vote_average={10}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "movie_id": 2,
          "poster_path": "/qIODP7buZlHBbbs6PNlQxb4orHo.jpg",
          "release_date": "2005-08-28",
          "starColor": "#ffd24d",
          "title": "Hotori - The Simple Wish for Joy",
          "vote_average": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(MovieCard)
            movie_id={2}
            poster_path="/qIODP7buZlHBbbs6PNlQxb4orHo.jpg"
            release_date="2005-08-28"
            starColor="#ffd24d"
            title="Hotori - The Simple Wish for Joy"
            vote_average={10}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "movie_id": 2,
            "poster_path": "/qIODP7buZlHBbbs6PNlQxb4orHo.jpg",
            "release_date": "2005-08-28",
            "starColor": "#ffd24d",
            "title": "Hotori - The Simple Wish for Joy",
            "vote_average": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`FavoritesList should match the snapshot if there are no favorites to display 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesList
    favorites={Array []}
    movies={
        Array [
            Object {
              "movie_id": 2,
              "poster_path": "/qIODP7buZlHBbbs6PNlQxb4orHo.jpg",
              "release_date": "2005-08-28",
              "title": "Hotori - The Simple Wish for Joy",
              "vote_average": 10,
            },
            Object {
              "movie_id": 3,
              "poster_path": "/gimNITXaKUevTsbXNUFEI6bCqro.jpg",
              "release_date": "1982-04-30",
              "title": "Big Meat Eater",
              "vote_average": 10,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
